pkg/adapter/mtping/adapter.go

```diff
@@ -128,3 +128,13 @@ func (a *mtpingAdapter) Remove(ctx context.Context, source *v1beta2.PingSource)
 		a.entryidMu.Unlock()
 	}
 }
+
+func (a *mtpingAdapter) RemoveAll(ctx context.Context) {
+	a.entryidMu.Lock()
+	defer a.entryidMu.Unlock()
+
+	for key, id := range a.entryids {
+		a.runner.RemoveSchedule(id)
+		delete(a.entryids, key)
```

Comment:
IMO Don't delete individually 
after the loop
a.entryids= map[string]cron.EntryID

Reply:
why not.